### DISCUSSION TITLE
How to create theme in stylex

### DISCUSSION THREAD

**Discussion by tienlx97** [facebook/stylex#106] [Q&A] [answered]
- I don't know how to customize theme. I have 2 obj with different css variable
- How to add it to stylex . I saw that at [StyleXSheet](https://github.com/facebook/stylex/blob/main/packages/stylex/src/StyleXSheet.js#L344),  you declared 2 empty object. Why don't you create setter function theme to config theme . Then export event to change it

```jsx
export class StyleXSheet {

  // @param {Theme} [theme]
  setRootTheme(theme) {
    this.rootTheme = theme;
    this.inject()
  }

  // @param {Theme} [theme]
  setRootTheme(theme) {
    this.rootDarkTheme= theme;
    this.inject()
  }
}
```

**Comment by necolas** [comment, marked as answer]
Docs:
1. https://stylexjs.com/docs/learn/theming/defining-variables/
2. https://stylexjs.com/docs/learn/theming/creating-themes/

**Comment by nmn** [comment]
Please do not use `StyleXSheet`. It is legacy code that will be removed soon.
